### ISSUE TITLE
num_workers > 0 leading to OOM in non-IPython environment

### ISSUE THREAD

**Issue by toemm** [pytorch/pytorch#13850]
My code works perfectly in an IPython Jupyter Notebook environment. I'm training a GAN while asynchronous loading data with ImageFolder and DataLoader. `Num_workers` behaves as expected.

When I export my code to .py and execute it from VSCode inside my conda environment (where I started the notebook from) my memory quickly goes up until it reaches `OOM: RuntimeError: CUDA error: out of memory`. (RAM goes to 100% aswell).

I am using this syntax because I'm on windows: 

```
import torch

def main()
    for i, data in enumerate(dataloader):
        # do something here

if __name__ == '__main__':
    main()
```
Using `num_workers=0` and `num_workers=1` works but then my epoch time increases about 22x. Everything over 1 leads to the aforementioned OOMs.

Theres shouldn't be any issue because I'm just copying my Ipython code to VSCode and then run the .py file from there, why is this happening?

**System information**

PyTorch version: 0.4.1
Is debug build: No
CUDA used to build PyTorch: 9.2

OS: Microsoft Windows 10 Home
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 9.2.148
GPU models and configuration: GPU 0: GeForce RTX 2070
Nvidia driver version: 416.81
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] Could not collect
[conda] cuda92                    1.0                           0    pytorch
[conda] pytorch                   0.4.1           py37_cuda92_cudnn7he774522_1  [cuda92]  pytorch
[conda] torchvision               0.2.1                     <pip>

**Comment by toemm** [comment]
> Is there any code I can try here?

I managed to fix it by putting my **whole** training code into main().
Before that I was only doing the training loop in main while keeping other functions outside. This led to memory buildup as soon as more worker processed were spawned that executed the code that comes before main() repeatedly.

I'm not a Windows/Python guy so this wasn't obvious behaviour for me. I took the code snippet from the Windows FAQs from the Pytorch site: https://pytorch.org/docs/stable/notes/windows.html

Is this normal procedure to put the whole code into main() when using num_workers multiprocessing?
